lib/db.js

```diff
@@ -1,17 +1,96 @@
+var assert = require('assert');
 var sqlite3 = require('sqlite3').verbose();
 var uuid = require('node-uuid');
 var dbu = require('./dbutils');
 var extend = require('extend');
+var P = require('bluebird');
 
 function DB (client, options) {
-    //this.conf = options.conf;
-    //this.log = options.log;
+    this.conf = options.conf;
+    this.log = options.log;
     this.client = client;
 
     // cache keyspace -> schema
     this.schemaCache = {};
 }
 
+/**
+ * Wrap common internal request state
+ */
+function InternalRequest (opts) {
+    this.domain = opts.domain;
+    this.table = opts.table;
+    this.keyspace = opts.keyspace || dbu.keyspaceName(opts.domain, opts.table);
+    this.query = opts.query || null;
+    this.schema = opts.schema || null;
+    this.columnfamily = opts.columnfamily || 'data';
+}
+
+/**
+ * Construct a new InternalRequest based on an existing one, optionally
+ * overriding existing properties.
+ */
+InternalRequest.prototype.extend = function(opts) {
+    var req = new InternalRequest(this);
+    Object.keys(opts).forEach(function(key) {
+        req[key] = opts[key];
+    });
+    return req;
+};
+
+DB.prototype._makeInternalRequest = function (domain, table, query) {
+    var self = this;
+
+    var req = new InternalRequest({
+        domain: domain,
+        table: table,
+        query: query,
+        columnfamily: 'data'
+    });
+    var schemaCacheKey = JSON.stringify([req.keyspace, domain]);
+    req.schema = this.schemaCache[schemaCacheKey];
+
+    if (req.schema) {
+        return P.resolve(req);
+    } else {
+        var schemaQuery = {
+            attributes: {
+                key: 'schema'
+            },
+            limit: 1
+        };
+        var schemaReq = req.extend({
+            query: schemaQuery,
+            columnfamily: 'meta',
```

Comment:
In SQLite the term `columnfamily` doesn't make a lot of sense. I think you want to use `table` instead. A Cassandra keyspace corresponds to a SQLite database.